Bump python-airos to 0.2.4

CoMPaTech opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
Update airOS module to 0.2.4 from 0.2.1 hence updating the exceptions
- 0.2.4 adding the missing Enums and optional height to solve the reported issue
- 0.2.2 unified prefix of AirOS (instead of none)

https://github.com/CoMPaTech/python-airos/blob/main/CHANGELOG.md / https://github.com/compatech/python-airos/compare/v0.2.1...compatech:v0.2.4

Documentation has not been updated yet, as the LiteBeam is not confirmed working yet

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #149884 
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

CoMPaTech committed:
Bump airos version

specialadapter commented:
Cheers Tom! I will retest upon merge. You are an example to the FOSS community mate!

joostlek commented on tests/components/airos/snapshots/test_sensor.ambr:
```diff
@@ -447,6 +447,7 @@
     'capabilities': dict({
       'options': list([
         'ap_ptp',
+        'ap_ptmp',
```
You're missing translation keys for this

CoMPaTech replied:
Valid point, would the below make sense? (Keeping state length sensibly minimum)

```
+    AirOSSensorEntityDescription(
+        key="wireless_role",
+        translation_key="wireless_role",
+        device_class=SensorDeviceClass.ENUM,
+        value_fn=lambda data: data.wireless.mode.value.replace("-", "_").lower(),
+        options=WIRELESS_MODE_OPTIONS,
+    ),
     AirOSSensorEntityDescription(
         key="wireless_mode",
         translation_key="wireless_mode",
```

with

```
       "wireless_mode": {
         "name": "Wireless mode",
+        "state": {
+          "ap_ptp": "Point-to-Point",
+          "ap_ptmp": "Point-to-Multi-Point",
+          "sta_ptp": "Point-to-Point"
+        }
+      },
+      "wireless_role": {
+        "name": "Wireless role",
         "state": {
           "ap_ptp": "Access point",
+          "ap_ptmp": "Access point",
           "sta_ptp": "Station"
         }
       },
```

CoMPaTech replied:
Thought about it some more reading other integrations, commit will suggest `hub` as terminology

joostlek replied:
Also. We should not have strings that are the same in here. As in, when someone is automating they should now pick between "access point" and "access point" and they both trigger on different data. Which is horrible to automate with this

CoMPaTech replied:
Which is why I landed on 191aca5

```
          "ap_ptmp": "Access hub",
          "ap_ptp": "Access point",
          "sta_ptp": "Station"
```

joostlek replied:
I think we need to decide what would be the easiest for the user to understand, I think we could also do binary sensors for these. Up to you. It also kinda depends on what users will use this with. As in, will people automate on the fact what mode the device is in? Or that it is in PTP mode?

CoMPaTech replied:
Binary sensors might make sense or eventually when we start controlling in a switch. I'll remove 'm for now so we can continue the merge and we'll bring 'm back later on.

I'll add an another docs PR to appropriately remove wireless_mode

CoMPaTech committed:
Add missing ap_ptmp translation

CoMPaTech committed:
Ause airOS 8 documentation names

CoMPaTech committed:
Remove for now

CoMPaTech committed:
Bump CI (stopping for unrelated issues)

CoMPaTech committed:
Bump CI (stopping for unrelated issues)

CoMPaTech commented:
Recap:
 - Removed 'wireless_mode' sensor, will be re-added as (most likely) binary_sensors at a later stage
 - Added doc removal PR
 - Changes exception and class naming all prefixed
